### PR TITLE
`Programming exercises`: Fix Swift exercise creation

### LINKED ISSUE #6436: `Programming exercises`: package name is not replaced in directory names for Swift exercises
### Describe the bug

For Swift exercises the package directory is not replaced with the package name the user entered during exercise setup. Therefore the test cannot be executed.

### To Reproduce

1. Create a new Swift programming exercise.
2. In the online editor, go to the tests repository. Check the directory structure to the left.

### Expected behavior

The `${packageName}` placeholder is replaced by the package name.

### Screenshots

_No response_

### Which version of Artemis are you seeing the problem on?

6.1.2

### What browsers are you seeing the problem on?

Firefox

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by b-fein**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Changes affecting Programming Exercises
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Motivation and Context
Closes #6436

### Description
There are some template directories for Swift exercises that are wrongly detected to be files. Therefore copying fails for those. This PR fixes the ignore-mechanism that matches those files and ignores them in the copying process.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Create a new Swift programming exercise.
2. Go to the online editor: Check that the placeholders as shown in the screenshot are replaced in the three exercise repositories by their intended package name.

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### Test Coverage
A new test was added that checks the intended behaviour.

### Screenshots
before:
Screenshot 2023-04-11 at 20-34-40 Code Editor

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/FileService.java [hunk 1]
```diff
@@ -609,7 +608,8 @@ private Path applyFilenameReplacements(final Path originalTargetPath) {
      * @return True, if the path is assumed to be a file but actually points to a directory.
      */
     private boolean isIgnoredDirectory(final Path filePath) {
-        return IGNORED_DIRECTORIES.stream().anyMatch(filePath::endsWith);
+        final String path = filePath.getFileName().toString();
+        return IGNORED_DIRECTORY_SUFFIXES.stream().anyMatch(path::endsWith);
```
Comment (b-fein) [inline src/main/java/de/tum/in/www1/artemis/service/FileService.java:612]:
```suggestion
        final String filename = filePath.getFileName().toString();
        return IGNORED_DIRECTORY_SUFFIXES.stream().anyMatch(filename::endsWith);
```

**Comment by Strohgelaender** [comment]
I'm a bit confused looking at this PR, the changes look to me like small refactorings that would not change the behavior. Is the fix using `String:endsWith` instead of `Path:endsWith`? If yes, what is the difference here?

**Comment by b-fein** [comment]
> Is the fix using String:endsWith instead of Path:endsWith.

Yes, it is. Example: Path `p = some/path/with/some_file.name`

- `Path::endsWith` only checks if _full_ path elements match, i.e. `p` ends with `some_file.name`, or `with/some_file.name`, but not with `file.name`.
- By taking `f = p.filename().toString()` first, we now have got the usual suffix checks for strings on `some_file.name`, e.g. `f` ends with `file.name`.

**Review by Strohgelaender — APPROVED** [review]
Thanks for the explanation, the new logic makes sens and the other code improvements look good.

**Pull request merged**